src/pydvl/utils/utility.py

```diff
@@ -145,21 +149,30 @@ def _utility(self, indices: FrozenSet) -> float:
 
         x_train, y_train = self.data.get_training_data(list(indices))
         x_test, y_test = self.data.get_test_data(list(indices))
-        try:
-            self.model.fit(x_train, y_train)
-            score = float(self.scorer(self.model, x_test, y_test))
-            # Some scorers raise exceptions if they return NaNs, some might not
-            if np.isnan(score):
-                if self.show_warnings:
-                    warnings.warn(f"Scorer returned NaN", RuntimeWarning)
-                return self.default_score
-            return score
-        except Exception as e:
-            if self.catch_errors:
-                if self.show_warnings:
+        with warnings.catch_warnings():
+            if not self.show_warnings:
+                warnings.simplefilter("ignore")
+            try:
+                # Clone the model to avoid the possibility
+                # of reusing a fitted estimator
+                try:
+                    model = clone(self.model)
+                except TypeError:
+                    # This happens if the passed model is not an sklearn model
+                    # In this case, we just make a deepcopy of the model.
+                    model = clone(self.model, safe=False)
```

Comment:
I think the cloning could be optional (defaulting to True). In any case, we should document that it is done.

Reply:
True, it should be documented. Don't you think adding yet another argument to the already packed `Utility` constructor is too much?

Reply:
Done.